### ISSUE TITLE
Is it possible to store in data or somewhere else files or paths to them?

### ISSUE THREAD

**Issue by sperezconesa** [datreant/MDSynthesis#80]
Hello,
I am considering using MDSynthesis in my projects but I was wondering a couple of things:
-Is it possible to store in simulations files (or maybe only their paths) like you do dataframes, for example a dssp.pdf file that I make from a trajectory. 
-Can you store any arbitrary object in a sim, like a dictionary, a list or even a function?
Best regards and thank you,
Sergio

**Comment by orbeckst** [comment]
@sperezconesa development on MDSynthesis has largely halted – @dotsdl is not really maintaining it anymore. Instead we suggest to used datreant directly.

MDSynthesis was a good idea but it turned out to be more useful and easier to maintain to just abstract it more and make it datreant. @dotsdl can say more.

**Comment by kain88-de** [comment]
You can save everything that you can serialize in a sim object. In the end, they are just files on your hard drive. You can write objects to disk using the pickle module like this.

```python
with open(treant['some_file_name'], 'w') as fh:
    pickle.dump(some_dic, fh)
```

and reading 

```python
with open(treant['some_file_name']) as fh:
    some_dic = pickle.load(fh)
```

With pickle functions should be possible to serialize too.

**Comment by sperezconesa** [comment]
Thanks for the help! I will try to get into datreant. Is datreant still mantained?

**Comment by kain88-de** [comment]
Yes it is still maintained. We work on bugs. We do consider the library
feature complete though. So not many new things will appear.

**Comment by sperezconesa** [comment]
I am having problems pickling universe objects. How was this done in MDSynthesis, so I can do it in datreant?

**Comment by orbeckst** [comment]
Pickling universes is a hard problem, see https://github.com/MDAnalysis/mdanalysis/issues/878 (and a whole bunch of  [somewhat related issues](https://github.com/MDAnalysis/mdanalysis/issues?q=is%3Aissue+pickle+universe+is%3Aopen)) so it does not work yet in MDAnalysis. (But we would *really* like it to work...)

In MDS we just stored the topology and trajectory and made `Simulation.universe` a managed attribute that would do a `return mda.Universe(self.topology, self.trajectory)` every time it was accessed. This worked (if you didn't care about any changes to the universe) but also was a bit problematic when people used in their code `sim.universe` instead of assigning it to a variable`u = sim.universe` because the former would load the universe *every time*, with poor performance. (I suppose, we really should have made it a function `Simulation.get_Universe()` to make clear that it does something expensive... poor UI choice.)

But the bottom line is that your best approach at the moment is to store in your Treant `Universe.filename` and `Universe.trajectory.filename` and then rebuild the Universe when you need it.

Feel free to ask questions on the [MDAnalysis user list](https://groups.google.com/forum/#!forum/mdnalysis-discussion), people might also be able to give advice.

**Comment by orbeckst** [comment]
The lines in the MDS code https://github.com/datreant/MDSynthesis/blob/980e9dd4d8b8df076b621a9a391be6179112dc40/src/mdsynthesis/treants.py#L89-L109

**Comment by sperezconesa** [comment]
Ok, that clarifies things! 
Thanks, I am really liking the library :)
